realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -133,13 +134,27 @@ private boolean containsDot(String s) {
         return arr;
     }
 
+    /**
+     * Returns the column indices for the given field name. If a linked field is defined, the column index for
+     * each
+     *
+     * @param fieldDescription fieldName or link path to a field name.
+     * @param validColumnTypes Legal field type for the last field a
+     * @return
+     */
     // TODO: consider another caching strategy so linked classes are included in the cache.
-    private long[] getColumnIndices(String fieldName, ColumnType fieldType) {
+    private long[] getColumnIndices(String fieldDescription, ColumnType... validColumnTypes) {
```

Comment:
call of this method with second argument `null` should be fixed to `getColumnIndices(fieldName, (Object[]) null)` to avoid warning.